script.js

```diff
@@ -1,12 +1,160 @@
+    episodeNameNumElement.classList.add("episode-name-num-text");
+    episodeNameNumElement.textContent = episodeNameNum;
+
+    const imageContainer = document.createElement("div");
+    imageContainer.classList.add("episode-image-container");
+
+    const episodeImage = document.createElement("img");
+    episodeImage.classList.add("medium-img");
+    episodeImage.src = episodeImageSrc;
+
+    const episodeDescriptionTextContainer = document.createElement("div");
+    episodeDescriptionTextContainer.classList.add(
+      "episode-description-container"
+    );
+
+    const episodeDescriptionTextElement = document.createElement("p");
+    episodeDescriptionTextElement.classList.add("episode-description");
+    episodeDescriptionTextElement.innerHTML = episodeDescription;
+
+    // Update info about displayed episodes
+    searchSpan.textContent = `Displaying 0/${episodeList.length} episodes`;
+
+    // Add the episode card elements to the container
+    episodeNameNumHolder.appendChild(episodeNameNumElement);
+    episodeInfoCard.appendChild(episodeNameNumHolder);
+    episodeInfoCard.appendChild(imageContainer);
+    imageContainer.appendChild(episodeImage);
+    episodeDescriptionTextContainer.appendChild(episodeDescriptionTextElement);
+    episodeInfoCard.appendChild(episodeDescriptionTextContainer);
+    cardsContainer.appendChild(episodeInfoCard);
+  });
 }
 
-window.onload = setup;
+// level 200
+
+// function to search
+
+function displaySearchedEpisodes(inputValue) {
+  /*
+    allEpisodes [{}, {}, {}] => each episode is an object with props **the data from episodes.js**
+    arrayCollectionOfNameNumEpisode => an array of CURRENT nodes **from the dom**
+    each array element has access to .textContent
+  */
+
+  // creating a fresh array which only holds the episodes that match our search criteria
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    return (
+      episode.name.toLowerCase().includes(inputValue) ||
+      episode.summary.toLowerCase().includes(inputValue)
+    );
+  });
+
+  // Display searchedEpisodes
+  if (filteredEpisodes.length !== 0) {
+    rootElem.innerHTML = "";
+    makePageForEpisodes(filteredEpisodes);
+  }
+  searchSpan.textContent = `Displaying ${filteredEpisodes.length}/${allEpisodes.length} episodes`;
+}
+
+// List of event listeners (Should be always at the bottom)
+
+searchInput.addEventListener("input", (event) => {
+  const inputValue = event.target.value.trim().toLowerCase();
+  displaySearchedEpisodes(inputValue);
+});
+
+// level 300
+
+// jump to episode function
+
+function jumpToEpisode() {
+  const episodeSelectHTML = document.querySelector("#select-html"); // this is the dropdown
+  const position = episodeSelectHTML.value; // we set a variable for the value which is the number
+  const episodeSelectHTMLId = "episode-card" + position; // position is from the dropdown list; we are making the id will will find
+  document.getElementById(episodeSelectHTMLId).scrollIntoView();
```

Comment:
```suggestion
  document.getElementById(episodeSelectHTMLId).scrollIntoView({
    block: "center",
    behavior: "smooth",
  });
```
This should work. Header was overlapping because it's position is fixed. So now better

Reply:
Your solution is so elegant!